modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java

```diff
@@ -534,6 +534,11 @@ public static boolean needCastInSearchBounds(IgniteTypeFactory typeFactory, RelD
             return false;
         }
 
+        // No need to binary between char and varbinary.
+        if (SqlTypeUtil.isBinary(toType) && SqlTypeUtil.isBinary(fromType)) {
```

Comment:
this check is not operate with char types at all, why char is mentioned here ? also "No need to binary" is sounds strange, isn`t it ?

Reply:
That was a typo, thanks.

Reply:
I still think that such kind of fix is looks like a stub, varchar(default_precision) and varchar need to be equal types is not it ?

Reply:
What kind of fix we are talking about - you mean rewriting logic related to creating reldata types with default precision/scale?